polkadot/primitives/src/vstaging/mod.rs

```diff
@@ -445,30 +501,33 @@ pub fn skip_ump_signals<'a>(
 }
 
 impl CandidateCommitments {
-	/// Returns the core selector and claim queue offset determined by `UMPSignal::SelectCore`
-	/// commitment, if present.
-	pub fn core_selector(
-		&self,
-	) -> Result<Option<(CoreSelector, ClaimQueueOffset)>, CommittedCandidateReceiptError> {
+	/// Returns the ump signals of this candidate, if any, or an error if they violate the expected
+	/// format.
+	pub fn ump_signals(&self) -> Result<CandidateUMPSignals, CommittedCandidateReceiptError> {
+		let mut res = CandidateUMPSignals { select_core: None, approved_peer: None };
+
 		let mut signals_iter =
 			self.upward_messages.iter().skip_while(|message| *message != &UMP_SEPARATOR);
 
-		if signals_iter.next().is_some() {
-			let Some(core_selector_message) = signals_iter.next() else { return Ok(None) };
-			// We should have exactly one signal beyond the separator
-			if signals_iter.next().is_some() {
-				return Err(CommittedCandidateReceiptError::TooManyUMPSignals)
-			}
+		if signals_iter.next().is_none() {
+			// No UMP separator
+			return Ok(res)
+		}
 
-			match UMPSignal::decode(&mut core_selector_message.as_slice())
-				.map_err(|_| CommittedCandidateReceiptError::UmpSignalDecode)?
-			{
-				UMPSignal::SelectCore(core_index_selector, cq_offset) =>
-					Ok(Some((core_index_selector, cq_offset))),
-			}
-		} else {
-			Ok(None)
+		// Process first signal
+		let Some(first_signal) = signals_iter.next() else { return Ok(res) };
```

Comment:
Hmm, so we have a separator but then no signals. RFC103 defines no mandatory UMP signals. But then without any signals there is no good reason to have the separator at all. 

So I guess it is fine to still allow it here since the RFC doesn't make it clear.

Reply:
yes, it used to be the same prior to this PR